### ISSUE TITLE
Out of memory error if DEF FN with errors is followed by a correct one

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#19]
reported by Marcos Cruz https://sourceforge.net/p/pcbasic/discussion/bugs/thread/a6112b10/

**Comment by robhagemans** [comment]
Likely an issue with the recursion flag

**Comment by robhagemans** [comment]
15.08 only, works ok in 16.12

**Comment by robhagemans** [comment]
```
print fnz$(0)
Undefined user function
Ok
print fnz$(0)
Out of memory
Ok
```

**Comment by robhagemans** [comment]
fixed by commit a0f6ceb
